src/dataServices/dynamoDbBundleService.ts

```diff
@@ -314,9 +315,20 @@ export class DynamoDbBundleService implements Bundle {
             });
         } catch (e) {
             logger.error('Failed to lock', e);
+            for (let i = 0; i < lockedItems.length; i += 1) {
+                // if the request in the bundle was a PUT request, we have a race condition causing conflicts when locking/unlocking resources
+                if (lockedItems[i].isOriginalUpdateItem) {
+                    throw new ResourceConflictError(
```

Comment:
We can not be sure which item actually caused the conflict error here, there could be multiple update entries and only one of them is causing the conflict issue. Also, the Transaction method will unlock any successful locked items if the lock method return error information in a resolved promise [here](https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/pull/102/files#diff-bad4ab4404b21b69f94c2366d2b94daea20d791e5fd77ebeef937d45efa3b9c6R102), throwing an error will abort the Transaction method before it can unlock items.  

I think the following [block](https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/pull/102/files#diff-bad4ab4404b21b69f94c2366d2b94daea20d791e5fd77ebeef937d45efa3b9c6R329-R335) already covers this situation right?